src/planwise/client/projects2/components/common.cljs

```diff
@@ -0,0 +1,11 @@
+(ns planwise.client.projects2.components.common
+  (:require [planwise.client.dialog :refer [new-dialog]]
+            [planwise.client.components.common2 :as common2]))
+
+(defn delete-project-dialog
+  [state id]
+  [new-dialog {:open? @state
```

Comment:
It's more idiomatic to just pass the value of the `state` atom. You'd need to extract the cancel callback at the use point as well, but that change will decouple the component from the use case, which is a good thing.